From 5fdf47dd88d3174e6efbbb6563846f554c86f088 Mon Sep 17 00:00:00 2001
From: "Juan A. Suarez Romero" <jasuarez@igalia.com>
Date: Mon, 12 Sep 2016 16:06:22 +0000
Subject: [PATCH 16/32] i965/vec4: keep original type when dealing with null
 registers

Keep the original type when dealing with null registers. Specially
because we do no want to introduce an implicit conversion between
types that could affect the conditional flags.

This affects specially when the original type is DF, and we are working
on Ivybridge/Baytrail.
---
 src/mesa/drivers/dri/i965/brw_vec4.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/mesa/drivers/dri/i965/brw_vec4.cpp b/src/mesa/drivers/dri/i965/brw_vec4.cpp
index 4b9fee0024..9383e78fee 100644
--- a/src/mesa/drivers/dri/i965/brw_vec4.cpp
+++ b/src/mesa/drivers/dri/i965/brw_vec4.cpp
@@ -1996,6 +1996,7 @@ vec4_visitor::convert_to_hw_regs()
          case BAD_FILE:
             /* Probably unused. */
             reg = brw_null_reg();
+            reg.type = src.type;
             break;
 
          case MRF:
@@ -2046,6 +2047,7 @@ vec4_visitor::convert_to_hw_regs()
 
       case BAD_FILE:
          reg = brw_null_reg();
+         reg.type = dst.type;
          break;
 
       case IMM:
@@ -2218,6 +2220,8 @@ vec4_visitor::lower_simd_width()
          } else {
             if (inst->dst.file != ARF)
                dst = horiz_offset(inst->dst, channel_offset);
+            else
+               dst.type = inst->dst.type;
          }
          linst->dst = dst;
 
-- 
2.11.0

